rosys/vision/calibration.py

```diff
@@ -253,23 +253,24 @@ def project_from_image(self, image_coordinates: list[Point], target_height: floa
         """
 
     @overload
-    def project_from_image(self, image_coordinates: FloatArray, target_height: float = 0) -> FloatArray:
+    def project_from_image(self, image_coordinates: FloatArray, target_height: float = 0, frame: Frame3d | None = None) -> FloatArray:
         """Project points in image coordinates to a plane in world coordinates.
 
         :param image_coordinates: (Nx2) The image coordinates to project.
         :param target_height: The height of the plane in world coordinates.
+        :param frame: The coordinate frame from which to project the image coordinates to.
         :return: (Nx3) The world coordinates of the projected points.
         """
 
-    def project_from_image(self, image_coordinates: Point | list[Point] | FloatArray, target_height: float = 0) -> Point3d | None | list[Point3d | None] | FloatArray:
+    def project_from_image(self, image_coordinates: Point | list[Point] | FloatArray, target_height: float = 0, frame: Frame3d | None = None) -> Point3d | None | list[Point3d | None] | FloatArray:
         if isinstance(image_coordinates, Point):
             return self.project_from_image([image_coordinates], target_height=target_height)[0]
         if not isinstance(image_coordinates, np.ndarray):
             image_array = np.array([point.tuple for point in image_coordinates], dtype=np.float64)
             world_array = self.project_from_image(image_array, target_height=target_height)
             return [Point3d(x=point[0], y=point[1], z=point[2]) if not np.isnan(point).any() else None for point in world_array]
 
-        world_extrinsics = self.extrinsics.resolve()
+        world_extrinsics = self.extrinsics.resolve() if frame is None else self.extrinsics.transform_with(frame.resolve())
```

Comment:
I think you can just use:
```suggestion
        world_extrinsics = self.extrinsics.relative_to(frame)
```
Because `resolve` also just calls it with `None`. [Link](https://github.com/zauberzeug/rosys/blob/af186412fe38e4c10597927df2d72a7995be3ee5/rosys/geometry/object3d.py#L27-L45)